Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/common/ClusterManagerEx.java

```diff
@@ -155,26 +158,47 @@ public boolean isLIstAdditionalClusterSuccess() {
             return ClusterMetaDataService.getInstance().getCachedClusterDetails();
         }
 
-        List<SubscriptionDetail> subscriptionList = null;
-        try {
-            subscriptionList = manager.getSubscriptionManager().getSubscriptionDetails();
-            clusterDetails = ClusterManager.getInstance().getHDInsightClustersWithSpecificType(subscriptionList, OSTYPE);
-
-            // TODO: so far we have not a good way to judge whether it is token expired as we have changed the way to list hdinsight clusters
-            isListClusterSuccess = clusterDetails.size() != 0;
-        } catch (AggregatedException aggregateException) {
-            if (dealWithAggregatedException(aggregateException)) {
-                DefaultLoader.getUIHelper().showError("Failed to get HDInsight Cluster, Please make sure there's no login problem first","List HDInsight Cluster Error");
-            }
-        } catch (Exception ex) {
-            DefaultLoader.getUIHelper().showError("Failed to get HDInsight Clusters","List HDInsight Cluster Error");
-        }
+        Optional<List<SubscriptionDetail>> subscriptionList = manager.map(AzureManager::getSubscriptionManager)
+                .flatMap(subscriptionManager -> {
+                    try {
+                        return Optional.ofNullable(subscriptionManager.getSubscriptionDetails());
+                    } catch (IOException e) {
+                        DefaultLoader.getUIHelper().showError("Failed to get HDInsight Clusters. " +
+                                        "Please check your subscription and login at Azure Explorer.",
```

Comment:
>Azure Explorer [](start = 85, length = 14)

"View -> Tool Windows -> Azure Explorer"

Reply:
fix it